lib/strings.nix

```diff
@@ -1338,4 +1336,15 @@ rec {
         else if k == 2 then infixDifferAtMost2 ainfix binfix
         else levenshtein ainfix binfix <= k;
       in f;
-}
+} // (
+  let
+    head_tail_warn_msg = name:
+      ''
+        lib.strings.${name} only supports lists; use lib.lists.${name} instead.
+        The lib.strings.${name} alias is deprecated and will be removed after the NixOS 2024.11 release.
+      '';
```

Comment:
Imo it's a bit overkill to abstract such an error message, and as implemented this actually slows down `lib` evaluation a little bit, because of the extra `//` used. I think it would be fine to just inline the error twice

Reply:
Ah, I pulled it out because the main attrset is `rec` and already uses `head`/`tail` a few times but we can change the couple uses to explicit `builtins.{head,tail}` in the module

Reply:
I moved it into the main `rec {}` attrset with exports and changed occurrences of `head`/`tail` in the file to use `builtins.{head,tail}` explicitly